src/ArmTemplates/Extractor/Models/ExtractorParameters.cs

```diff
@@ -1,56 +1,133 @@
-        public string fileFolder { get; private set; }
-        public string linkedTemplatesBaseUrl { get; private set; }
-        public string linkedTemplatesSasToken { get; private set; }
-        public string linkedTemplatesUrlQueryString { get; private set; }
-        public string policyXMLBaseUrl { get; private set; }
-        public string policyXMLSasToken { get; private set; }
-        public string apiVersionSetName { get; private set; }
-        public bool includeAllRevisions { get; private set; }
-        public serviceUrlProperty[] serviceUrlParameters { get; set; }
-        public bool paramServiceUrl { get; private set; }
-        public bool paramNamedValue { get; private set; }
-        public bool paramApiLoggerId { get; private set; }
-        public bool paramLogResourceId { get; private set; }
-        public bool notIncludeNamedValue { get; private set; }
-        public bool paramNamedValuesKeyVaultSecrets { get; private set; }
-
-        public int operationBatchSize { get; private set; }
-
-        public bool paramBackend { get; set; }
-
-        public ExtractorParameters(ExtractorConfig exc, string dirName)
+        public string SourceApimName { get; private set; }
+
+        public string DestinationApimName { get; private set; }
+
+        public string ResourceGroup { get; private set; }
+
+        /// <summary>
+        /// Naming of all templates from the base folder requested by end-user in extractor-configuration parameters
+        /// </summary>
+        public FileNames FileNames { get; private set; }
+
+        /// <summary>
+        /// Not changeable root directory, where templates will be generated.
+        /// </summary>
+        public string FilesGenerationRootDirectory { get; private set; }
+
+        /// <summary>
+        /// Name of a single API that user wants to extract
+        /// </summary>
+        public string SingleApiName { get; set; }
+
+        /// <summary>
+        /// Names of APIs that user wants to extract
+        /// </summary>
+        public List<string> MultipleApiNames { get; set; }
+
+        /// <summary>
+        /// Create split api templates for every API in the source APIM instance
+        /// </summary>
+        public bool SplitApis { get; private set; }
+
+        /// <summary>
+        /// Are revisions included to templates
+        /// </summary>
+        public bool IncludeAllRevisions { get; private set; }
+
+        public string LinkedTemplatesBaseUrl { get; private set; }
+
+        public string LinkedTemplatesSasToken { get; private set; }
+
+        public string LinkedTemplatesUrlQueryString { get; private set; }
+
+        public string PolicyXMLBaseUrl { get; private set; }
+
+        public string PolicyXMLSasToken { get; private set; }
+
+        public string ApiVersionSetName { get; private set; }
+
+        public ServiceUrlProperty[] ServiceUrlParameters { get; set; }
+
+        public bool ToParameterizeServiceUrl { get; private set; }
```

Comment:
Can you explain these to me please? Why do they start with a "To" and what does Parametrize mean?

Reply:
Briefly looking at the rest of the code, I still don't understand why it's called Parametrize (pls help me understand), but I think we can drop the "To".

Reply:
It is like **ServiceUrl** is needed **To** be **Parameterized**. In another way we can say, then we (as end-users) want **To Parameterize Service Url**. (_describe or represent in terms of a parameter or parameters_)

I am trying to give a somehow understandable name to property, because before it was `ParamServiceUrl`. What is done with service url is absolutely not recognisable from such a naming. In one file I have found such a naming with fully written word **parameterize** and decided to go with that by now.

I do not understand right now what Extractor does with service url, maybe will rename it later.